src/ui/common/src/utils/storage.ts

```diff
@@ -34,3 +36,102 @@ export type StorageConfig = {
   file_config?: FileConfig;
   gcs_config?: GCSConfig;
 };
+
+export type MetadataStorageConfig = {
+  aqPath: string;
+  retentionJobPeriod: string;
+  apiKey: string;
+  storageConfig: {
+    type: StorageType;
+    fileConfig?: {
+      directory: string;
+    };
+    gcsConfig?: {
+      bucket: string;
+    };
+    s3Config?: {
+      region: string;
+      bucket: string;
+    };
+  };
+};
+
+export async function getMetadataStorageConfig(
```

Comment:
This is the same as the GetServerConfig route. Could you use that instead?

Reply:
I think you're talking about `getServerConfig` in `src/ui/common/src/components/pages/account/AccountPage.tsx`. I moved it to this file and am using this function for `AccountPage` and `deleteIntegrationDialog`.

Reply:
Shoot, I have a PR out here which refactors this into an asyncThunk:
https://github.com/aqueducthq/aqueduct/pull/1009

We'll unify these two code paths to use the thunk in a future task.

Reply:
[https://linear.app/aqueducthq/issue/ENG-2607/refactor-use-asyncthunk-to-retrieve-server-config-on-frontend](url)

Reply:
Nice! I integrated it since the change has been merged to main already.